Added some env variables handling

ViktorLindgren95 opened the pull request:
I started working on handling some environment variables so that when we eventually move to docker / k8s we are ready for it. Take a look and give an opinion of these changes before i merge then into the schema branch please

Kind regards V

Commit:
Added Env handling for most of our environment secrets. also some tests

felixlut commented on handlerGithub/github.go:
```diff
@@ -114,15 +126,20 @@ func convertToGithubRepositorySchema(repositoryPayload github.RepositoryPayload)
-	createSonarQubeFile(repositoryPayload)
-	//	githubRepository.Components.Sonarqube = *sonarQubeComponent
+	if os.Getenv("enable-sonarqube") == "true" { // we could check the token but i think thats the sonarqube libraries job
+		sonarQubeComponent, err2 := sonarqube.OnboardSonarQube(repositoryPayload)
```
`err2` ?!?

felixlut replied:
Fair enough. Add a linter to enforce no shadowing of errors? And maybe not name them `err`, `err2`, ... 
Maybe just my ocd talking, but that is ugly as shit imo

ViktorLindgren95 replied:
whatr err inner or what? error errrroror haha

ViktorLindgren95 commented:
To avoid shadowing .

On Sun, Dec 4, 2022, 12:02 Felix Luthman ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In handlerGithub/github.go
> <https://github.com/FridayCommit/Cherios/pull/3#discussion_r1038950022>:
>
> >  	}
> -	// there should be an if statement here reading if the component is enabled
> -	//	sonarQubeComponent, err := sonarqube.OnboardSonarQube(repositoryPayload)
> -	//	if err != nil {
> -	//		return nil, err
> -	//	}
> -	createSonarQubeFile(repositoryPayload)
> -	//	githubRepository.Components.Sonarqube = *sonarQubeComponent
> +	if os.Getenv("enable-sonarqube") == "true" { // we could check the token but i think thats the sonarqube libraries job
> +		sonarQubeComponent, err2 := sonarqube.OnboardSonarQube(repositoryPayload)
>
> err2 ?!?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FridayCommit/Cherios/pull/3#pullrequestreview-1203635243>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFRJY4FTKTMZWCJHGQ4XR6DWLR24VANCNFSM6AAAAAASQVG3PM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

felixlut commented on handlerGithub/github.go:
```diff
@@ -58,10 +66,14 @@ type Components struct {
 	//	if len(appKey) < 1 {
 	//		log.Fatalln("Missing App Key")
 	//	}
+	appID, err := strconv.ParseInt(os.Getenv("appid"), 10, 64)
```
Any reason to re-read and parse the env to `appID` on every init of the gh-client? Why not read it once like `repoAsCodeOrg` and the other global vars?

felixlut replied:
Or maybe read `appID` before calling `initGitHubClient`, add pass it as an argument

ViktorLindgren95 replied:
Could setup a function that checks the environment variables and sets them accordingly i guess. might be bad code quality hmm

ViktorLindgren95 replied:
Oh read it in main function, Yeah sure we can do that.

ViktorLindgren95 replied:
Passing it sounds sounds a lot better

ViktorLindgren95 replied:
I will setup an init function that sorts this out. This way we can also check mandatory values :)

ViktorLindgren95 replied:
I played with the idea of creating a function in main that checks all the environment variables but thats bad. I believe that the best practice is either having an init function for the module that checks the environment variables or not even setting vars but instead calling each time.

ViktorLindgren95 replied:
Or we just let the function that needs these variables fail because they will fail when you pass an empty argument i think

ViktorLindgren95 commented:
Because that one calls a function with multiple return values , i don't
know how to do it as a global :P

On Sun, Dec 4, 2022, 12:07 Felix Luthman ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In handlerGithub/github.go
> <https://github.com/FridayCommit/Cherios/pull/3#discussion_r1038950697>:
>
> >  	//	if len(appKey) < 1 {
>  	//		log.Fatalln("Missing App Key")
>  	//	}
> +	appID, err := strconv.ParseInt(os.Getenv("appid"), 10, 64)
>
> Any reason to re-read and parse the env on every init of the gh-client?
> Why read it once like repoAsCodeOrg and the other global vars?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FridayCommit/Cherios/pull/3#pullrequestreview-1203635867>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFRJY4CQ2G4YLKK5GYTXZN3WLR3Q5ANCNFSM6AAAAAASQVG3PM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ViktorLindgren95 commented:
That's a good idea

On Sun, Dec 4, 2022, 12:10 Felix Luthman ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In handlerGithub/github.go
> <https://github.com/FridayCommit/Cherios/pull/3#discussion_r1038951041>:
>
> >  	//	if len(appKey) < 1 {
>  	//		log.Fatalln("Missing App Key")
>  	//	}
> +	appID, err := strconv.ParseInt(os.Getenv("appid"), 10, 64)
>
> Or maybe read appID before calling initGitHubClient, add pass it as an
> argument
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FridayCommit/Cherios/pull/3#discussion_r1038951041>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFRJY4EDWLHWYNFPHU3ADJDWLR33NANCNFSM6AAAAAASQVG3PM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

felixlut reviewed (commented):
Code quality questions

Commit:
Added readme
Made the decision to make conversions on the function level

Commit:
According to tests my idea didnt work.

So conversion for single use is done on function level
reused things are fetched in Init of the usage function. right now that is createSourcehook

Commit:
According to tests my idea didnt work.

So conversion for single use is done on function level
reused things are fetched in Init of the usage function. right now that is createSourcehook

ViktorLindgren95 committed:
Merge branch 'schema' into schema-env
